Add readme and contributor file to UI (#142)
open-edge-platform/geti: web_ui/README.md

For a full deep dive visit our developer documentation [`here`](https://docs.geti.intel.com/developer-guide/interactive-ai/frontend/).

## Application structure

The web UI of Geti is a Single Page Application utilizing [react router](https://reactrouter.com/) for client side routing. [Rsbuild](https://rsbuild.dev) is our build tool.
Below is a diagram of our route structure,

![geti-route-structure](https://github.com/user-attachments/assets/baaae7a0-af5b-4d1c-8fa1-4ae8a71a8a38)

The `intel-admin` routes are used to manage organizations and users as well as assigning credits. These routese are used by the [admin environment](https://github.com/open-edge-platform/geti/blob/7e46430b7afeca6288f7e751d49adacac61fd7f3/web_ui/rsbuild.config.ts#L66-L77) while the other routes are used by the [geti environment](https://github.com/open-edge-platform/geti/blob/7e46430b7afeca6288f7e751d49adacac61fd7f3/web_ui/rsbuild.config.ts#L53-L65).
The admin environment isn't available via our installer or helm chart instructions, this will likely change in the future.

> Keep reading to get an overview of our core architecture, our stack, testing strategies and how we use client side computer vision algorithms to speed up annotation time.

Reviewer: ```suggestion
Keep reading to get an overview of our core architecture, stack, testing strategies and usage of client side computer vision algorithms to speed up annotation time.
```

## Architecture 

> - **Core stack**: Our UI is a React application build using Rsbuild & npm. Typescript is language of choice.

Reviewer: ```suggestion
- **Core stack**: Our UI is a React application build using Rsbuild & npm. Typescript is our language of choice.
```

- **Application**: React Router and TanStack Query are critical for managing navigation and data fetching efficiently in our application.
- **Testing**: we aim to have a high code coverage where we use jest for unit tests and Playwright for our component and end-to-end tests.
- **Algorithms & AI**: Our UI contains smart annotation tools that speed up annotation efforts. We use [OpenCV](https://opencv.org/), [OnnxRUntime](https://onnxruntime.ai) compiled to [WebAssembly](https://webassembly.org/) and use [WebWorkers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) to make sure heavy computions don't block the UI thread.

---

### Core stack

| ![npm](./npm.png) | ![Typescript](./typescript.png) | ![React](./react.png) | ![Rsbuild](./rsbuild.png) |
|-------------------|---------------------------------|-----------------------|---------------------------|

Our core architecture consists of using [npm](https://www.npmjs.com/) as our package manager, [Typescript](https://www.typescriptlang.org/) as the language of choice and [react](react.dev) as our frontend framework.

> Initially the UI was build from [Create React App](https://github.com/facebook/create-react-app). We've migrated since to [rsbuild](https://rsbuild.dev/). 

Reviewer: ```suggestion
Initially the UI was build from [Create React App](https://github.com/facebook/create-react-app). Which we then migrated to [rsbuild](https://rsbuild.dev/). 
```

### Testing

| ![Testing Library](./testing-library.png) | ![Jest](./jest.png) | ![Playwright](./playwright.png) |
|-------------------------------------------|---------------------|---------------------------------|

- **Avoiding a test user**: We use [Testing Library](https://testing-library.com/)'s interfaces both in our unit and component/e2e tests. This allows us to avoid having a [testing user](https://kentcdodds.com/blog/avoid-the-test-user). This also aligns well with React Aria's focus on accessibility.
- **Unit tests**: We use [jest](https://jestjs.io/) as our unit testing framework.
- **Playwright**: Our component (isolated UI browser tests that rely on a mocked backend) and End-to-End (No mocking, tests are run on an actual server) use [Playwright](https://playwright.dev/)

### Algorithms & AI

| ![WebAssembly](./web-assembly.png) | ![OpenCV](./open-cv.png) | ![ONNXRuntime](./onnx-runtime.png) | ![Web Workers](./web-workers.png) |
|------------------------------------|--------------------------|------------------------------------|------------------------------------|

> Our annotator contains smart tools such as [Detection Assitant](https://docs.geti.intel.com/docs/user-guide/geti-fundamentals/annotations/annotation-tools#detection-assistant-tool) or [Automatic Segmentation](https://docs.geti.intel.com/docs/user-guide/geti-fundamentals/annotations/annotation-tools#automatic-segmentation-tool) are local only and rely on [Web Workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) to keep the UI smooth and non blocking when we are performing heavy compute.

Reviewer: np: typo "assitant"
